libs/server/Storage/Functions/MainStore/RMWMethods.cs

```diff
@@ -671,6 +695,27 @@ public bool CopyUpdater(ref SpanByte key, ref SpanByte input, ref SpanByte oldVa
                     rmwInfo.Action = RMWAction.ExpireAndStop;
                     return false;
 
+                case RespCommand.GETEX:
+                    CopyRespTo(ref oldValue, ref output);
+
+                    if (input.ExtraMetadata > 0)
+                    {
+                        Debug.Assert(newValue.Length == oldValue.Length + input.MetadataSize);
+                        byte* pbOutput = stackalloc byte[ObjectOutputHeader.Size];
+                        var _ = new SpanByteAndMemory(SpanByte.FromPinnedPointer(pbOutput, ObjectOutputHeader.Size));
+                        EvaluateExpireCopyUpdate(ExpireOption.None, expiryExists: oldValue.MetadataSize > 0, ref input, ref oldValue, ref newValue, ref _);
+                    }
+
+                    oldValue.AsReadOnlySpan().CopyTo(newValue.AsSpan());
+                    var isPersist = *(bool*)(inputPtr + RespInputHeader.Size);
```

Comment:
Is this necessary? If we're only doing GET + PERSIST, would we ever reach CopyUpdate?

Reply:
Ideally not, add it just in case, if you are sure then i can remove it. Shall I?

Reply:
Yeah I think this should be removed...

Reply:
Ok, I will remove it, Do you know why the existing `PERSIST` comment also has an entry in CopyUpdater?

Reply:
Fixed it

Reply:
Now I'm doubting myself too :) Maybe if it needs to move somewhere else in memory... Let me double check with someone in the team.

Reply:
I added it back, but do you think we need to add the if condition back in the `GetRMWModifiedValueLength` from the other review comment?

Reply:
If the source of the RMW is on disk or in the immutable region of memory, we have to issue a CU (CopyUpdater) even though the size does not increase. So yes, CopyUpdate would be needed.

Reply:
@badrishc Thanks for the explanation, it makes sense. The developer doc should be updated with this info. Now it just says, "you will need to implement the initialization and in-place / copy update functionality of the new command"

Reply:
Fixed